src/intermediate_terms.jl

```diff
@@ -0,0 +1,180 @@
+# * Seniority
+
+"""
+    Seniority(ν)
+
+Seniority is an extra quantum number introduced by Giulio Racah (1943)
+to disambiguate between terms belonging to a subshell with a specific
+occupancy, that are assigned the same term symbols. For partially
+filled f-shells (in ``LS`` coupling) or partially filled ``9/2``
+shells (in ``jj`` coupling), seniority alone is not enough to
+disambiguate all the arising terms.
+
+The seniority number is defined as the minimum occupancy number `ν ∈
+n:-2:0` for which the term first appears, e.g. the ²D term first
+occurs in the d¹ configuration, then twice in the d³ configuration
+(which will then have the terms ₁²D and ₃²D).
+"""
+struct Seniority
+    ν::Int
+end
+
+Base.isless(a::Seniority, b::Seniority) = a.ν < b.ν
+Base.iseven(s::Seniority) = iseven(s.ν)
+Base.isodd(s::Seniority) = isodd(s.ν)
+
+# This is the notation by Giulio Racah, p.377:
+# - Racah, G. (1943). Theory of complex spectra. III. Physical Review,
+#   63(9-10), 367–382. http://dx.doi.org/10.1103/physrev.63.367
+Base.show(io::IO, s::Seniority) = write(io, to_subscript(s.ν))
+
+istermvalid(term, s::Seniority) =
+    iseven(multiplicity(term)) ⊻ iseven(s)
+
+# This is due to the statement "For partially filled f-shells,
+# seniority alone is not sufficient to distinguish all possible
+# states." on page 24 of
+#
+# - Froese Fischer, C., Brage, T., & Jönsson, P. (1997). Computational
+#   Atomic Structure : An Mchf Approach. Bristol, UK Philadelphia, Penn:
+#   Institute of Physics Publ.
+#
+# This is too strict, i.e. there are partially filled (ℓ ≥ f)-shells
+# for which seniority /is/ enough, but I don't know which, so better
+# play it safe.
+assert_unique_classification(orb::Orbital, occupation, term::Term, s::Seniority) =
+    istermvalid(term, s) &&
+    (orb.ℓ < 3 || occupation == degeneracy(orb))
+
+function assert_unique_classification(orb::RelativisticOrbital, occupation, J::HalfInteger, s::Seniority)
+    ν = s.ν
+    # This is deduced by looking at Table A.10 of
+    #
+    # - Grant, I. P. (2007). Relativistic Quantum Theory of Atoms and
+    #   Molecules : Theory and Computation. New York: Springer.
+    istermvalid(J, s) &&
+        !((orb.j == half(9) && (occupation == 4 || occupation == 6) &&
+           (J == 4 || J == 6) && ν == 4) ||
+          orb.j > half(9)) # Again, too strict.
+end
```

Comment:
Did I miss it or we currently do not have `assert_unique_classification` for the `S = Int` case?

Reply:
I mean, what would that be? Since we have not devised the integer scheme yet, and now `Seniority` is its own type, I don't know what a valid, integer-only, classification would be.

Reply:
Just that it should be in `1:count_terms(...)`. But we can leave this for now.